File: src/homework05/services/EnumSelector.java
package homework05.services;

import homework05.mvp.View;

import java.util.Arrays;
import java.util.List;

public interface EnumSelector {
    static <T> T selectFromInput(View view, Class<T> enumClass) {
        List<T> enumList = Arrays.asList(enumClass.getEnumConstants());
        String enumName = enumClass.getSimpleName();
        int length = enumList.size();
        while (true) {
            StringBuilder sb = new StringBuilder();
            sb.append(String.format("Choose %s:%n", enumName));
            for (int i = 0; i < length; i++) {
                sb.append(String.format("%d - %s%n", i, enumList.get(i)));
            }
            view.showMenu(sb.toString());
            int actionId = view.getIntegerFieldFromInput();
            if (actionId < 0 || actionId > length) continue;
            return enumList.get(actionId);
        }
    }
}

File: src/homework07/phonebook/services/enum_selector.py
class EnumSelector:
    @staticmethod
    def select_from_input(view, enum_class):
        enum_len = len(enum_class)
        enum_menu = f'Choose {enum_class.__name__}:\n'
        enum_menu += ''.join(f'{i} - {v.name}\n' for i, v in enumerate(enum_class))
        enum_list = list(enum_class)
        while True:
            view.show_menu(enum_menu)
            action_id = view.get_int_from_input()
            if action_id < 0 or action_id > enum_len:
                continue
            return enum_list[action_id]
